subsys/modem/modem_cmux.c

```diff
@@ -876,12 +877,16 @@ static void modem_cmux_transmit_handler(struct k_work *item)
 		ret = modem_pipe_transmit(cmux->pipe, reserved, reserved_size);
 		if (ret < 0) {
 			ring_buf_get_finish(&cmux->transmit_rb, 0);
+			if (cmux->pipe->state != MODEM_PIPE_STATE_CLOSED) {
```

Comment:
The modem_pipe_transmit API returns `-EPERM` if the pipe is closed, I can see this documentation is missing from the `modem_pipe.h`, it is only mentioned for the receive function... could you add this documentation, and check the return value instead of directly reading the internal context of the pipe? The pipe context is private, and should not be accessed directly :)

Reply:
Okay, did as you suggested. However how about the hypothetical case where `pipe->api->transmit` (or `receive`, for `modem_pipe_receive()` would return `-EPERM`?